Merge .h and .cc files of plugins in several PhysicsTools packages

guitargeek opened the pull request:
#### PR description:

Part of the series of PRs that merges `.h` files with the `.cc` files of the plugins in the PhysicsTools subsystem, continuing the work from https://github.com/cms-sw/cmssw/pull/34594.

It makes maintaining the plugins much easier and also excludes the possibility to wrongly include plugin header files.

There were the steps taken to merge the files:
* copy-paste
* sort includes
* removed duplicate and unnecessary includes

#### PR validation:

CMSSW compiles.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

guitargeek committed:
Merge .h and .cc files of some more PhysicsTools plugins

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34615/24192

- This PR adds an extra 28KB to repository

cmsbuild commented:
A new Pull Request was created by  @guitargeek (Jonas Rembser) for master.

It involves the following packages:

- PhysicsTools/IsolationAlgos (**reconstruction**)
- PhysicsTools/PatAlgos (**reconstruction**)
- PhysicsTools/TagAndProbe (**analysis**)


@perrotta, @jpata, @cmsbuild, @santocch, @slava77 can you please review it and eventually sign? Thanks.
@mmarionncern, @gouskos, @JyothsnaKomaragiri, @ahinzmann, @schoef, @rappoccio, @swozniewski, @jdamgov, @jdolen, @nhanvtran, @gkasieczka, @clelange, @emilbols, @hatakeyamak, @mbluj, @gpetruc, @andrzejnovak, @mariadalfonso, @seemasharmafnal this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy, @perrotta you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-49923b/17178/summary.html
**COMMIT**: 0673c6ea32f88a734e05b2d00e10cb03bb0073c4
**CMSSW**: CMSSW_12_0_X_2021-07-25-0000/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/34615/17178/install.sh` to create a dev area with all the needed externals and cmssw changes.

**CMS StaticAnalyzer warnings**: There are 1 inherits from legacy modules warnings. See https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-49923b/17178/llvm-analysis/legacy-mod-sa.txt for details.

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 2998564
 - DQMHistoTests: Total failures: 12
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2998529
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 38 files compared)
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 165 log files, 37 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-25-0000+49923b/44334/triggerResults): no differences found

jpata reviewed (commented):
Thanks for this refactoring! Please see inline for a few questions.

jpata commented:
kind ping @guitargeek, just in case you are around

jpata commented:
kind ping @guitargeek

jpata commented:
kind ping @guitargeek
there's also a conflict.

guitargeek commented on PhysicsTools/PatAlgos/plugins/TauJetCorrFactorsProducer.cc:
```diff
@@ -130,26 +126,24 @@ std::vector<JetCorrectorParameters> TauJetCorrFactorsProducer::params(
   // get tau-jet collection from the event
-  edm::Handle<edm::View<reco::BaseTau> > tauJets;
-  evt.getByToken(srcToken_, tauJets);
+  auto tauJets = evt.getHandle(srcToken_);
```
Okay right, but this commit is outdated now because the esConsumes migration of the `TauJetCorrFactorsProducer` was done in https://github.com/cms-sw/cmssw/pull/34997.

guitargeek commented on PhysicsTools/PatAlgos/plugins/TauJetCorrFactorsProducer.cc:
```diff
@@ -162,37 +156,31 @@ void TauJetCorrFactorsProducer::produce(edm::Event& evt, const edm::EventSetup&
-      es.get<JetCorrectionsRecord>().get(payload, jecParameters);
-
-      correctorMapping[payload] = std::make_shared<FactorizedJetCorrector>(params(*jecParameters, levels_));
+    const int payloadTokenValue = payloadToken->index().value();
```
Okay right, but this commit is outdated now because the esConsumes migration of the `TauJetCorrFactorsProducer` was done in https://github.com/cms-sw/cmssw/pull/34997.

guitargeek commented:
Hi @jpata, sorry for the late response! I have fixed the conflict now and dropped my second commit with the esConsumes migration of the `TauJetCorrFactorsProducer` plugin, since this migration has now already been done in https://github.com/cms-sw/cmssw/pull/34997. What's left in this PR is the merging of .h and .cc files of plugins in several PhysicsTools packages, which should be easy to review.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34615/24979

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #34615 was updated. @jpata, @cmsbuild, @santocch, @slava77 can you please check and sign again.

jpata commented:
@cmsbuild please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-49923b/18188/summary.html
**COMMIT**: d363c3b37d28dbf5e59d43012b7e1827af301576
**CMSSW**: CMSSW_12_1_X_2021-08-31-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34615/18188/install.sh` to create a dev area with all the needed externals and cmssw changes.

**CMS StaticAnalyzer warnings**: There are 1 inherits from legacy modules warnings. See https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-49923b/18188/llvm-analysis/legacy-mod-sa.txt for details.

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 7 differences found in the comparisons
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 3000404
 - DQMHistoTests: Total failures: 11
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3000370
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 38 files compared)
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 165 log files, 37 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_1_X_2021-08-31-2300+49923b/45123/triggerResults): no differences found

jpata commented:
+reconstruction
- technical
- reco differences in ParticleNet and DeepTau look like from https://github.com/cms-sw/cmssw/issues/32628 and are unrelated, nothing else observed

perrotta commented:
+1

perrotta commented:
merge

santocch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will be automatically merged.
